src/ngraph/pass/reshape_sinking.cpp

```diff
@@ -175,6 +175,20 @@ void swim(Input<Node> input, shared_ptr<op::Reshape> reshape)
         auto csw = work_queue.front();
         work_queue.pop_front();
         auto n = csw.input.get_source_output().get_node_shared_ptr();
+
+        auto materialize = [csw, n]() {
+            auto new_reshape = csw.reshape->copy_with_new_args({n});
+            new_reshape->merge_provenance_tags_from(n);
+            NGRAPH_DEBUG << "Materializing new reshape " << describe_reshape(new_reshape);
+            csw.input.replace_source_output(new_reshape->output(0));
+        };
+
+        // Only swim past nodes which have a single user
+        if (n->get_users(true) > 1)
```

Comment:
`== 1`?

Reply:
I had it this way with a big conditional around the swimming if/elsif block but it made the github diff huge and hid the change. It can be productized however if @ashokei confirms that it helps.

Reply:
build issue... `n->get_users(true).size()` ? 

seems to fix the issue though reshape cant be avoided, backend can optimize further.